src/frontend/configuration/node_migration.rs

```diff
@@ -0,0 +1,563 @@
+use crate::frontend::configuration::utils::{calculate_node_data_size, get_available_space};
+use crate::frontend::translations::{AsDefaultStr, T};
+use crate::frontend::NODE_FREE_SPACE_WARNING_THRESHOLD;
+use crate::icon_names;
+use bytesize::ByteSize;
+use gtk::glib;
+use gtk::prelude::*;
+use relm4::prelude::*;
+use relm4_components::open_dialog::{
+    OpenDialog, OpenDialogMsg, OpenDialogResponse, OpenDialogSettings,
+};
+use std::path::PathBuf;
+use tracing::warn;
+
+/// Migration mode selection
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Default)]
+pub enum MigrationMode {
+    /// Copy the existing database to new location
+    #[default]
+    Migrate,
+    /// Fresh sync from network at new location
+    FreshSync,
```

Comment:
Ideal to be specific here?
We have snap sync and full sync. If we want to support snap sync first, the it should be communicated accordingly